src/corecel/data/Collection.hh

```diff
@@ -127,6 +127,41 @@ using ItemId = OpaqueId<T, size_type>;
 template<class T, class Size = size_type>
 using ItemRange = Range<OpaqueId<T, Size>>;
 
+//---------------------------------------------------------------------------//
+/*! Access data in a Range<T2> with an index of type T1
+ *
+ * Here, T1 and T2 are expected to be OpaqueIds.
+ */
+template<class T1, class T2>
```

Comment:
I wasn't sure where I should put this. Let me know and I will write tests for it as well. Also should an `ItemMap` implement all of the methods of `Range`?

Reply:
Okay I will write a few tests as well unless I hear otherwise